### DISCUSSION TITLE
Home-brew won't install

### DISCUSSION THREAD

**Discussion by iliyan61** [Homebrew/discussions#145] [Everyday usage]
Hi I'm trying to install home-brew on a fresh install and it keeps erroring out.
Here's the error lines

`Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 1295, done.
remote: Counting objects: 100% (1295/1295), done.
remote: Compressing objects: 100% (729/729), done.
error: RPC failed; curl 18 transfer closed with outstanding read data remaining
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-core /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core` exited with 128.
Error: Failure while executing; `/usr/local/bin/brew tap homebrew/core` exited with 1.`

**Comment by SMillerDev** [comment]
Your network is abandoning the connection before it can complete the transfer. There is not much I can do to help you with this, check your firewall/virusscanner/network to try and work around it.

**Reply by iliyan61** [reply]
there's no reason it shouldn't be working and I was able to clone a regular git repo. Is there any way I can test it? The settings are all defaults and on Friday I installed home-brew on a clean install that was the same settings.

**Reply by gromgit** [reply]
> there's no reason it shouldn't be working

"The network is reliable" is first among the [fallacies of distributed computing](https://en.wikipedia.org/wiki/Fallacies_of_distributed_computing). The fact that "on Friday I installed home-brew on a clean install that was the same settings" is no guarantee that it'll work perfectly today, on a different machine, possibly even on a different network path.

> Is there any way I can test it?

The best test is to redo the cloning by hand, but first check [GitHub's own status page](https://www.githubstatus.com/) to see if they're having any issues with Git Operations.

Then, if `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/.git/config` exists:
```
git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core pull
```
otherwise, as the error message says:
```
git clone https://github.com/Homebrew/homebrew-core /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
```
`homebrew-core` is the largest Homebrew repo by far, so once you've cloned it successfully, run the installer script once more, and it should complete gracefully.

**Reply by Amasiani** [reply]
having the same error while trying to reinstall Homebrew after it was installed in the wrong PATH in my MacOs m1 silicon machine. 

Cloning into '/opt/homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 1101427, done.
remote: Counting objects: 100% (48/48), done.
remote: Compressing objects: 100% (28/28), done.
error: RPC failed; curl 18 transfer closed with outstanding read data remaining
error: 3593 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: index-pack failed
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-core /opt/homebrew/Library/Taps/homebrew/homebrew-core --origin=origin --template=` exited with 128.

**Comment by jeevanism** [comment]
I have the below isse when I tried to install HomeBrew as per the instruction from website : https://brew.sh/

`==> Tapping homebrew/core
Cloning into '/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 1027881, done.
error: RPC failed; curl 18 transfer closed with outstanding read data remaining
error: 1376 bytes of body are still expected
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
Error: Failure while executing; `git clone https://github.com/Homebrew/linuxbrew-core /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core` exited with 128.
Error: Failure while executing; `/home/linuxbrew/.linuxbrew/bin/brew tap homebrew/core` exited with 1.
Failed during: /home/linuxbrew/.linuxbrew/bin/brew update --force
`

**Reply by gromgit** [reply]
Same issue, same solution set.

**Reply by jeevanism** [reply]
I dont see a solution first, I am not sure if the OP got it solved or not. I tried this first - https://www.githubstatus.com/ and yes everything green to go.  and I done the
```
 jeevanism@jeevanism-81d2:~/homebrew> git clone https://github.com/Homebrew/homebrew-core
Cloning into 'homebrew-core'...
remote: Enumerating objects: 216, done.
remote: Counting objects: 100% (216/216), done.
remote: Compressing objects: 100% (113/113), done.
error: RPC failed; curl 18 transfer closed with outstanding read data remaining
error: 806 bytes of body are still expected
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
```
So again I went this page : https://brew.sh/ and run this command 
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"`
then  I got 
`Error: Failure while executing; `git clone https://github.com/Homebrew/linuxbrew-core /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core` exited with 128.`

I dont see a solution anywhere. Anyway thankx

**Reply by SMillerDev** [reply]
https://github.com/Homebrew/discussions/discussions/145#discussioncomment-131017

**Reply by gromgit** [reply]
You'll have to:
1. Ensure that any proxies, firewalls, anti-virus, etc. aren't interfering with `git clone` to GitHub, as @SMillerDev said before.
1. Read my instructions _very carefully_. (I said "as the error message says". _Your_ error message specifies a different `git clone` command than @iliyan61's.)
1. Keep retrying the clone until it succeeds, then re-run the installer (which should finally run with no errors). Homebrew's "database" is basically a set of Git repos, of which `homebrew-core` is the largest by far. Once you've successfully cloned that, the others should not be an issue.

**Comment by MattMcFahn** [comment]
If anyone has this issue - SMillerDev was correct in that the Network was the issue for me. I wasted a couple hours trying different routes before fixing this!

If anyone has this issue and they're on Virgin Media, their ISP settings have anti virus built in which was the issue for me. You can manage your settings here: 

https://my.virginmedia.com/my-apps/onlinesecurity/websafe/settings

I added https://github.com as an allowed website, and it finally worked. I think there might be similar issues with some other ISPs.

**Reply by iliyan61** [reply]
yeh im pretty sure virgin media is super fucky with git

**Reply by alibvr** [reply]
Thanks a lot. it works for me

**Reply by paulberhane** [reply]
Spot on Matt, I had same issue today and after reading your comment, managed to fix it. Kudos!

**Reply by stukennedy** [reply]
fixed it for me too ... very irritating

**Reply by seagreenman** [reply]
And another Virgin Media user who has fixed this issue with Matt's solution - thanks!

**Reply by myanch200** [reply]
@MattMcFahn **Thank you** saved me a lot of nerves 👍

**Reply by matlo607** [reply]
Thanks a lot 👍 I have been struggling for a day with this issue before finding your comment.

**Reply by ghost** [reply]
Thanks Matt.I am facing this issue from last 6 hrs.Did everything what people advised.Finally found your solution. I changed the network and connected via mobile hotspot.It has resolved. Definitely it has an issue with VirginMediaProvider. Never thought virgin media was the culprit here.

**Reply by biswajitchakraborty** [reply]
This was spot on and it fixed for me too. Thanks @MattMcFahn

**Reply by bradleythomasporter** [reply]
Same! Thank you!

**Reply by Lhamide00** [reply]
Still saving lives in 2022 @MattMcFahn  thanks!!

**Comment by richie-r** [comment]
Hi guys, I have just had the same issue on a new macbook pro. I thought it was maybe Monterey, but when i saw this post I changed my DNS servers to Googles (8.8.8.8 & 8.8.8.4) and made sure that the battery settings didn't put the laptop to sleep and hey presto, it's working!

**Reply by Antony343** [reply]
Looks like laptop should not fall into sleep mode while downloading for some reason.

**Reply by yeunsoo53** [reply]
This is what my problem was too. Once I made sure it didn't go on sleep mode while downloading, it worked.

**Reply by qshiho11** [reply]
amazing!!! why this way can solve that bothered issue???
